dll/win32/shell32/dialogs/folder_options.cpp

```diff
@@ -1443,20 +1599,50 @@ InsertFileType(HWND hDlgCtrl, WCHAR * szName, PINT iItem, WCHAR * szFile)
             RegQueryValueExW(hKey, L"EditFlags", NULL, NULL, (LPBYTE)&Entry->EditFlags, &dwSize);
     }
 
+    /* convert extension to upper case */
+    wcscpy(Entry->FileExtension, szName);
+    _wcsupr(Entry->FileExtension);
+
+    /* get icon */
+    if (!GetFileTypeIconsByKey(hKey, Entry))
+    {
+        WCHAR szDefaultIconLocation[] = L"%SystemRoot%\\system32\\SHELL32.dll,-210";
+        GetFileTypeIconsEx(Entry, szDefaultIconLocation);
```

Comment:
Can we instead use some resource loading using shell32_instance HINSTANCE instead of hardcoding that path?

Reply:
OK.